python/cli/data_platform_workflows_cli/update_bundle.py

```diff
@@ -24,36 +25,45 @@ def get_ubuntu_version(series: str) -> str:
     ).stdout.split(" ")[0]
 
 
-def fetch_latest_revision(charm, charm_channel, series=None) -> int:
-    """Gets the latest charm revision number in channel."""
+def fetch_charm_info_from_store(charm, charm_channel) -> Tuple[Dict, Dict]:
+    """Returns, for a given channel, the necessary charm info from store endpoint."""
     response = requests.get(
-        f"https://api.snapcraft.io/v2/charms/info/{charm}?fields=channel-map"
+        f"https://api.snapcraft.io/v2/charms/info/{charm}?fields=channel-map,default-release&channel={charm_channel}"
     )
     response.raise_for_status()
     channel_map = response.json()["channel-map"]
-    track, risk = charm_channel.split("/")
+    metadata = yaml.safe_load(response.json()["default-release"]["revision"]["metadata-yaml"])
+    return channel_map, metadata
+
+
+def fetch_latest_revision(channel_map, series=None) -> int:
+    """Gets the latest charm revision number in channel."""
     revisions = []
     for channel in channel_map:
         if (
-            channel["channel"]["risk"] == risk
-            and channel["channel"]["track"] == track
```

Comment:
no need to check for channel here, we no fetch charm's data by channel directly from store.